### PR TITLE
Enable download of EAD file

### LINKED ISSUE #434: Add ability to download finding aid as EAD

### PR DESCRIPTION
**Pull request by corylown**
Closes #434 

Because this relies on the filename being stored in Solr we'll need to re-index after this is merged.

### CODE REVIEW

**Comment by marlo-longley** [comment]
Hey Cory -- I tried this locally and got errors so I reindexed as suggested in the comment.

Then I got this error after clicking Download EAD for ARS-0043:

Screenshot 2024-04-05 at 3 36 43 PM

**Comment by marlo-longley** [comment]
@corylown never mind -- I set the DATA_DIR variable and this worked. I wonder if we should set `DATA_DIR=spec/fixtures/ead` by default somewhere for development.

It works though! Nice.

File: app/controllers/download_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# Controller for downloading finding aid files
+class DownloadController < ApplicationController
+  def show
+    doc = SolrDocument.find(params[:id])
+
+    data_dir = ENV.fetch('DATA_DIR', Rails.root.join('data').to_s)
```
Comment (marlo-longley) [inline app/controllers/download_controller.rb:8]: For me this folder doesn't exist`Rails.root.join('data').to_s` so it had an unhandled error, but maybe that's fine.

Comment (corylown) [reply app/controllers/download_controller.rb:8]: @marlo-longley I guess this is due to me typically using the `DownloadEadJob` to index finding aids and that task by default stores data in the `./data` directory. Sounds like you're usually using the fixture data. Maybe we should add the `./data` directory to the repo. I'm a little stumped about the right way to address this. Maybe a setting, but again I'm not sure if the default should be `./data` or `./spec/fixtures/ead`.

Comment (marlo-longley) [reply app/controllers/download_controller.rb:8]: Hm maybe we can have it all and make the default /data if it exists, then /spec/fixtures/ead?

Comment (corylown) [reply app/controllers/download_controller.rb:8]: Sigh...I think switching to ActiveStorage to manage the files would solve this for us. Maybe I'll see where I can get with that.

Comment (marlo-longley) [reply app/controllers/download_controller.rb:8]: Ah I was not trying to cause a spiral with this comment! We could just commit the data directory or write this out for local dev instructions. Obviously on the deployed servers the ENV is set and the directory exists.

Comment (corylown) [reply app/controllers/download_controller.rb]: @marlo-longley made a couple adjustments, including making the fixtures directory the default for development for now.

**Review by marlo-longley — APPROVED** [review]

**Pull request merged**